keyless/pepper/service/src/lib.rs

```diff
@@ -129,6 +137,138 @@ impl HandlerTrait<PepperRequest, SignatureResponse> for V0SignatureHandler {
+
+                            // The training wheels signature is only checked if a training wheels PK is set on chain
+                            if training_wheels_pk.is_some() {
+                                match &zksig.training_wheels_signature {
+                                    Some(training_wheels_sig) => {
+                                        training_wheels_sig
+                                            .verify(
+                                                &groth16_and_stmt,
+                                                training_wheels_pk.as_ref().unwrap(),
+                                            )
+                                            .map_err(|_| {
+                                                // println!("[aptos-vm][groth16] TW sig verification failed");
+                                                invalid_signature!(
+                                                    "Could not verify training wheels signature"
+                                                )
+                                            })?;
+                                    },
+                                    None => {
+                                        // println!("[aptos-vm][groth16] Expected TW sig to be set");
+                                        return Err(invalid_signature!(
+                                            "Training wheels signature expected but it is missing"
+                                        ));
+                                    },
+                                }
+                            }
+
+                            let result = zksig
+                                .verify_groth16_proof(public_inputs_hash, &DEVNET_VERIFICATION_KEY);
+                            result.map_err(|_| {
+                                // println!("[aptos-vm][groth16] ZKP verification failed");
+                                // println!("[aptos-vm][groth16] PIH: {}", public_inputs_hash);
+                                // match zksig.proof {
+                                //     ZKP::Groth16(proof) => {
+                                //         println!("[aptos-vm][groth16] ZKP: {}", proof.hash());
+                                //     },
+                                // }
+                                // println!(
+                                //     "[aptos-vm][groth16] PVK: {}",
+                                //     Groth16VerificationKey::from(pvk).hash()
+                                // );
+                                invalid_signature!("Proof verification failed")
+                            })?;
+                        },
+                    }
+                },
+                EphemeralCertificate::OpenIdSig(_) => {
+                    return Err(invalid_signature!(
+                        "Could not verify training wheels signature"
+                    ))
+                },
+            }
+        }
```

Comment:
rn vrfy passes if the pk and skg are not keyless. but i wonder if you actually mean to fail vrfy in this case

Reply:
u right.  thought i was using match, not if let